spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/config/DocumentDbConfigurationSupport.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.azure.spring.data.documentdb.config;
+
+import com.microsoft.azure.spring.data.documentdb.core.mapping.DocumentDbMappingContext;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.data.annotation.Persistent;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.data.mapping.context.MappingContextIsNewStrategyFactory;
+import org.springframework.data.mapping.context.PersistentEntities;
+import org.springframework.data.support.CachingIsNewStrategyFactory;
+import org.springframework.data.support.IsNewStrategyFactory;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
```

Comment:
From performance standpoint its always better to use static imports instead of util.*